src/transaction/log_recovery.c

```diff
@@ -884,6 +882,129 @@ log_recovery (THREAD_ENTRY * thread_p, int ismedia_crash, time_t * stopat)
+ *                    transaction table information at the time of the crash and finalize transactions;
+ *                    this starts with the last available transaction table snapshot and performing recovery
+ *                    analysis from that LSA followed by undo
+ *
+ * return: nothing
+ *
+ */
+void
+log_recovery_finish_transactions (THREAD_ENTRY * const thread_p)
+{
+  assert (is_tran_server_with_remote_storage ());
+  assert (log_Gl.m_metainfo.is_loaded_from_file ());
+
+  const int sys_tran_index = LOG_FIND_THREAD_TRAN_INDEX (thread_p);
+  assert (sys_tran_index == LOG_SYSTEM_TRAN_INDEX && LOG_FIND_TDES (sys_tran_index) != nullptr);
+
+  // TODO: logging has been ignored ?
+
+  // recovery analysis phase
+  //
+  log_Gl.rcv_phase = LOG_RECOVERY_ANALYSIS_PHASE;
+
+  // start with what metalog tells us
+  const auto[chkpt_lsa, chkpt_info] = log_Gl.m_metainfo.get_highest_lsa_checkpoint_info ();
+  if (chkpt_info == nullptr)
+    {
+      logpb_fatal_error (thread_p, true, ARG_FILE_LINE,
+			 "log_recovery_finish_transactions: no checkpoint found; control should not end up here");
+      // unreachable
+      return;
+    }
+  assert (!chkpt_lsa.is_null ());
+
+  log_recovery_context log_rcv_context;
+  // TODO: is media crash check?
+  log_rcv_context.init_for_recovery (chkpt_lsa);
+
+  long redo_log_record_count = 0;
+  log_recovery_analysis (thread_p, &redo_log_record_count, log_rcv_context);
+  er_set (ER_NOTIFICATION_SEVERITY, ARG_FILE_LINE, ER_LOG_RECOVERY_STARTED, 3,
+	  (long long) redo_log_record_count, log_rcv_context.get_start_redo_lsa ().pageid,
+	  log_rcv_context.get_end_redo_lsa ().pageid);
+  // analysis changes the transaction index and leaves it dangling in an indefinite state
+  // therefore reset to system transaction index
+  LOG_SET_CURRENT_TRAN_INDEX (thread_p, sys_tran_index);
+
+  // make sure the append page is available
+  // needed to read the lsa of the last added log record that is found there
+  if (logpb_fetch_start_append_page (thread_p) != NO_ERROR)
+    {
+      logpb_fatal_error (thread_p, true, ARG_FILE_LINE,
+			 "log_recovery_finish_transactions: logpb_fetch_start_append_page");
+      // unreachable
+      return;
+    }
+
+  // read the end of file record to find out about the previous address
+  assert (log_Gl.append.log_pgptr != nullptr);
+  assert (!log_Gl.hdr.append_lsa.is_null ());
+  assert (!log_rcv_context.is_restore_incomplete ());
+  const LOG_RECORD_HEADER *const log_append_log_header =
+    (const LOG_RECORD_HEADER *) (log_Gl.append.log_pgptr->area + log_Gl.hdr.append_lsa.offset);
+  LOG_RESET_PREV_LSA (&log_append_log_header->back_lsa);
+
+  assert (!log_Gl.append.prev_lsa.is_null ());
+  vacuum_notify_server_crashed (&log_Gl.append.prev_lsa);
+
+  log_recovery_abort_all_atomic_sysops (thread_p);
+
+  log_recovery_finish_all_postpone (thread_p);
+
+  const int err_code_reset_db_parm = boot_reset_db_parm (thread_p);
```

Comment:
Using a specific constant error code variable for each function call is just too much work. Having one error code variable is a pattern repeated in many functions; it easy to understand and to followed by other coders.

```suggestion
  int error_code = boot_reset_db_parm (thread_p);
```

Reply:
```suggestion
  // Reset boot_Db_parm in case a data volume creation was undone.
  const int err_code_reset_db_parm = boot_reset_db_parm (thread_p);
```